GitFyle.Core.Api.Tests.Unit/Services/Foundations/Sources/SourceServiceTests.Validations.Modify.cs

```diff
@@ -0,0 +1,438 @@
+        {
+            //given
+            var invalidSource = new Source
+            {
+                Name = invalidString,
+                Url = invalidString,
+                UpdatedBy = invalidString,
+            };
+
+            var invalidSourceException = new InvalidSourceException(
+                message: "Source is invalid, fix the errors and try again.");
+
+            invalidSourceException.AddData(
+                key: nameof(Source.Id),
+                values: "Id is invalid");
+
+            invalidSourceException.AddData(
+                key: nameof(Source.Name),
+                values: "Text is required");
+
+            invalidSourceException.AddData(
+                key: nameof(Source.Url),
+                values: "Url is invalid");
+
+            invalidSourceException.AddData(
+                key: nameof(Source.CreatedBy),
+                values: "Text is required");
+
+            invalidSourceException.AddData(
+                key: nameof(Source.UpdatedBy),
+                values: "Text is required");
+
+            invalidSourceException.AddData(
+                key: nameof(Source.CreatedDate),
+                values: "Date is invalid");
+
+            invalidSourceException.AddData(
+                key: nameof(Source.UpdatedDate),
+                values:
+                    new[]
+                    {
+                        "Date is invalid",
+                        $"Date is the same as {nameof(Source.CreatedDate)}"
+                    });
+
+            var expectedSourceValidationException =
+                new SourceValidationException(
+                    message: "Source validation error occurred, fix errors and try again.",
+                    innerException: invalidSourceException);
+
+            // when
+            ValueTask<Source> modifySourceTask =
+                this.sourceService.ModifySourceAsync(invalidSource);
+
+            SourceValidationException actualSourceValidationException =
+                await Assert.ThrowsAsync<SourceValidationException>(
+                    modifySourceTask.AsTask);
+
+            // then
+            actualSourceValidationException.Should().BeEquivalentTo(
+                expectedSourceValidationException);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffsetAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogErrorAsync(It.Is(
+                    SameExceptionAs(expectedSourceValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertSourceAsync(It.IsAny<Source>()),
+                    Times.Never);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
```

Comment:
Add the test for `ShouldThrowValidationExceptionOnModifyIfSourceHasInvalidLengthPropertiesAndLogItAsync`.   You can reuse the one done on the Add and just modify this.

Reply:
Added